src/System.Transactions.Local/src/System/Transactions/InternalTransaction.cs

```diff
@@ -164,41 +164,16 @@ internal DistributedTransaction PromotedTransaction
 
         // Object for synchronizing access to the entire class( avoiding lock( typeof( ... )) )
         private static object s_classSyncObject;
-        internal static object ClassSyncObject
-        {
-            get
-            {
-                if (s_classSyncObject == null)
-                {
-                    object o = new object();
-                    Interlocked.CompareExchange(ref s_classSyncObject, o, null);
-                }
-                return s_classSyncObject;
-            }
-        }
 
         internal Guid DistributedTxId => State.get_Identifier(this);
 
         // Double-checked locking pattern requires volatile for read/write synchronization
```

Comment:
>volatile [](start = 51, length = 8)

I am not clear on why the volatile keyword can be removed here. How does LazyInitializer.EnsureInitialized avoid requiring it to be volatile?

If it is, indeed, safe to remove volatile, then this comment should be removed.